### ISSUE TITLE
AioZipStream does not support async generators

### ISSUE THREAD

**Issue by mojimi** [kbbdy/zipstream#3]
The following example does not work :

´´´python
        async def files_generator():
            for filename in ['test1.txt', 'test2.txt', 'test3.txt']:
                yield {
                    'stream': await storage.get_file_stream({'path' : filename}, 5),
                    'name': filename
                }

        zs = AioZipStream(files_generator())
´´´

**Comment by kbbdy** [comment]
Hi. I can't say what's wrong without knowing what is `storage.get_file_stream` method and what it returns. You also not added any error log, so I can't help. Remember that stream generator should return chunks of binary data.

**Comment by kingmad** [comment]
@kbbdy `AioZipStream` does not support async generator as an input. Changing `for` loop to `async for` in `AioZipStream.stream#78` will probably do the job in this particular case

**Comment by kbbdy** [comment]
Ok, thanks. I will check that.

**Comment by mojimi** [comment]
Yes exactly what kingmad said, sorry I should've explained it better.

Basically by not working with an async generator it won't free up memory and/or the asyncio loop in some cases

**Comment by ChristianMeisenbichler** [comment]
I would also like to use an async iterator for the source of the files list maybe implement a  stream_async_iter method ? Something where the loop looping over the files (`AioZipStream.stream#78`) is `async for`?
